### PR TITLE
doc/user: revamp `CREATE SOURCE` landing page

### LINKED ISSUE #7885: doc: Source documentation restructure
### Motivation

The organization of our docs for sources has been a long-standing pet peeve. On one hand, there is an unnecessary amount of information duplication, and on the other the pages seem to grow wild, with no real structure that can help users navigate the content.

We should rip off the band-aid and do a full restructure that can improve not only the user experience, but also the docs maintainability.

### Proposal

**1. Create `connect/sources.md` page**

Instead of directly linking to `/sql/create-source/`, create a new `/connect/sources.md` landing page that gives users a quick-to-parse overview of all the supported sources, including high-level information about **connectors**, **formats** and **envelopes**. In contrast, `/sql/create-source` should provide lower-level information, like connector->format mapping and direct links to the dedicated connector pages.

**2. Create `/sql/create-source/<connector>.md` pages**

The current page structure is hard to parse and not intuitive to navigate. There is also an When landing on a connector page, it should be **immediate** to users:

* Formats and envelopes supported.

* Desirable features supported (_e.g._ for Kafka, surfacing "Using source metadata", "Setting start offsets", "Handling upserts" as submenu levels instead of having all the different supported features jammed under "Kafka source details"), including examples with the reference syntax. For desirable features that we explicitly know are not supported, we should still list them and link to the relevant tracking tickets, to gauge user interest and avoid repeated questions about the same set of features (_e.g._ enhanced fan-out in Kinesis Data Streams). Non-generic `WITH` options should also be associated with the respective relevant features (_i.e._ filled under the matching submenu), instead of bulked into an ever-growing list with all the _possible_ options.

* Authentication options.

**3. Adapt railroad diagrams**

Each existing connector+format page uses a subset of the railroad diagram specific to the format. This is probably the most annoying and time consuming part to sort out, though [`format spec`](https://materialize.com/docs/sql/create-source/protobuf-kafka/#format_spec) already seems to conflate the syntax for all formats.

In the future, I'd like to explore an **alternative** to the current syntax representation.

## Changelog

Touching all source docs will inevitably lead to some intense yak shaving, as it is a golden opportunity to incorporate issues that have been laying around for a while. Below is a changelog of documented and undocumented issues that will be covered as part of the restructuring.

#### Kafka + Redpanda (fixed in #10678)

~- [x] https://github.com/MaterializeInc/materialize/issues/9611~
~- [x] https://github.com/MaterializeInc/materialize/pull/9643~
~- [x] https://github.com/MaterializeInc/materialize/issues/7645~
~- [x] https://github.com/MaterializeInc/materialize/pull/9997 / https://github.com/MaterializeInc/materialize/issues/10257~

~**Undocumented issues**~

~- [x] Replace `mz_kafka_consumer_partitions` with `mz_kafka_source_statistics` for the `statistics_interval_ms` configuration option (related to #8904).~
~- [x] [Basic HTTP authentication](https://docs.confluent.io/platform/current/security/basic-auth.html#basic-auth-sr) for schema registry is supported, but undocumented.~

#### Kinesis Data Streams (fixed in #10914)

- [ ] https://github.com/MaterializeInc/materialize/issues/9652 (also applies to S3)
~- [x] https://github.com/MaterializeInc/materialize/issues/9616~

~**Undocumented issues**~

~- [x] Replace mentions to Rusoto’s `ChainProvider` with AWS Rust SDK (#8754)~

#### S3 (fixed in #10966)

~- [x] https://github.com/MaterializeInc/materialize/pull/9936~

~**Undocumented issues**~

~- [ ] Replace mentions to Rusoto’s `ChainProvider` with AWS Rust SDK (related to #8754)~

#### PostgreSQL (fixed in #10528)

~- [x] https://github.com/MaterializeInc/materialize/issues/10219~

~**Undocumented issues**~

~- [ ] Creating views with a different schema to that of the upstream source is undocumented (see [this thread](https://materializeinc.slack.com/archives/C01CFKM1QRF/p1637684697353100))~
~- [x] Be more intentional about explaining source-specific details (_e.g._ why does the source have to be materialized?) and ensure that some FAQs are answered (see #7281).~

### Formats

- [x] https://github.com/MaterializeInc/materialize/issues/9935
- [ ] https://github.com/MaterializeInc/materialize/issues/7885

@benesch, I'd love to hear your thoughts on what kind of blurb we should provide for formats (you seemed to have some specific wants!). What would, for example, the section about Avro touch? Would it make sense to include _e.g._ supported primitive types, or something like a format->Materialize data type mapping?

### Envelopes

**Undocumented issues**

- [ ] De-emphasize `MATERIALIZE` envelope (see [this discussion](https://materializeinc.slack.com/archives/C015RHB3LDR/p1639476737167800)). Remove `/connect/materialize-cdc.md` and instead link to this page under the envelope description in `connect/sources.md`. 

<hr>

Final TODO:

- [ ] Update [developer docs](https://github.com/MaterializeInc/materialize/tree/main/doc/user)

👹

### PR DESCRIPTION
**Pull request by morsapaes**
Closes #7885.

👉 **[Preview link](https://preview.materialize.com/12880/sql/create-source/)**

### Tips for reviewer

The new page describes formats and envelopes in a way that can answer common questions raised by users. The original idea was to include details like data type mappings, but this ended up feeling out of place in such an overview-ish page. If we want to include this kind of detail, we could add full-on format-specific pages later on (which could also provide more low-level information in the style of [this write-up](https://umanwizard.com/avro.rst)).

@benesch, if you could give a closer look to the `Avro` and `Protobuf` sections, it'd be great! I'm not super familiar with the lingo, so might have typed nonsense.

### CODE REVIEW

**Review by benesch — APPROVED** [review]

File: doc/user/content/sql/create-source/_index.md [hunk 1]
```diff
@@ -1,44 +1,231 @@
...
-Don't see what you're looking for? [Let us know on GitHub](https://github.com/MaterializeInc/materialize/issues/new?labels=C-feature&template=feature.md).
+To read from an external data source, Materialize must be able to determine how to decode raw bytes from different formats into data structures it can understand at runtime. This is handled by specifying a `FORMAT` in the `CREATE SOURCE` statement.
+
+### Avro
+
+<p style="font-size:14px"><b>Syntax:</b> <code>FORMAT AVRO</code></p>
+
+Materialize can decode Avro messages by integrating with a schema registry to retrieve a schema, and automatically define the columns and data types to use in the source.
+
+##### Schema versioning
```
[the remark this answers is not in the document]
Comment (morsapaes) [reply doc/user/content/sql/create-source/_index.md:50]: I did, because H4 looks an awful lot like H3, which made the page confusing. :marta: But at some point there was a `#### Format details` header preceding these; I just...didn't like how unnecessary it looked.

File: doc/user/content/sql/create-source/_index.md [hunk 2]
```diff
@@ -1,44 +1,231 @@
...
     identifier: 'create-source'
     weight: 20
 ---
 
-`CREATE SOURCE` connects Materialize to some data source, and lets you interact
-with its data as if it were in a SQL table.
+A [source](../../overview/key-concepts/#sources) describes an external system you want Materialize to read data from, and provides details about how to decode and interpret that data. To create a source, you must specify a [connector](#connectors), a [format](#formats) and an [envelope](#envelopes).
 
-## Conceptual framework
+Executing a `CREATE SOURCE` statement does **not** prompt data ingestion; it simply registers source metadata in the [system catalog](/sql/system-catalog/#mz_catalog). Data ingestion will only kick off when you create a materialization that depends on an existing source, such as a [materialized view](../../overview/key-concepts/#materialized-views).
```
[the remark this answers is not in the document]
Comment (morsapaes) [reply doc/user/content/sql/create-source/_index.md:15]: Ugh, my Cloud brain didn't fully form yet; I'll adapt it here but keep it for the backport. Thanks!

**Comment by morsapaes** [comment]
Thanks, @benesch! I'll give this another pass once I'm a little more settled on how to explain the new behaviour. Merging for now!
